satellite/repair/repairer/repairer.go

```diff
@@ -110,7 +110,7 @@ func (service *Service) worker(ctx context.Context, seg *pb.InjuredSegment) (err
 	workerStartTime := time.Now().UTC()
 
 	service.log.Info("Limiter running repair on segment", zap.Binary("segment", seg.GetPath()))
-	err = service.repairer.Repair(ctx, string(seg.GetPath()))
+	shouldDelete, err := service.repairer.Repair(ctx, string(seg.GetPath()))
```

Comment:
Make sure you still check `shouldDelete` if `err != nil`

Reply:
wait, really? normal convention is all guarantees about other return values are off if err != nil. that's surprising

Reply:
maybe to make it clear we're bucking convention, it should be `err, shouldDelete := service.repairer.Repair(...`?

Reply:
oh, I guess there are certain errors that we will want to delete for, and other errors that we don't want to delete for. But we can keep it as is and explicitly delete if the error is a certain type rather than relying on `shouldDelete`. I don't know the best practice in this situation.

Reply:
I guess this should be an exception from the normal convention similar to `Read`: https://golang.org/pkg/io/#Reader
In `Read` the caller is supposed to check the `n` return value even if `err != nil`.

Reply:
I added a comment clarifying the use of shouldDelete